### PR TITLE
Proposition for network modification builder

### PR DESCRIPTION
**Pull request by miovd**
Signed-off-by: VEDELAGO MIORA <miora.ralambotiana@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Improvement

**What is the current behavior?** *(You can also link to an open issue here)*
A constructor (sometimes not really user-friendly) is used to create a network modification.

**What is the new behavior (if this is a feature change)?**
We can add a builder for each modification as proposed. If the design is agreed upon, it can be implemented for other network modifications.

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [x] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

**Comment by geofjamg** [comment]
Good idea.
Also theses modifications are really badly named. I would suggest to rename AttachNewLineOnLine to InsertLineIntoLine and AttachVoltageLevelOnLine to InsertVoltageLevelIntoLine

File: iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/AttachNewLineOnLine.java [hunk 1]
```diff
@@ -54,7 +54,9 @@ public class AttachNewLineOnLine implements NetworkModification {
      * line2Id is line.getId() + "_2". <br>
      *
      * @see #AttachNewLineOnLine(double, String, String, String, String, String, Line, LineAdder)
+     * @deprecated Use {@link AttachNewLineOnLineBuilder} instead.
      */
+    @Deprecated(since = "4.10.0")
```
Comment (flo-dup) [inline iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/AttachNewLineOnLine.java:59]: Just wondering if deprecated is the appropriate term / tag, as we will just change the access to protected (and not really delete it one day).

**Review by flo-dup — APPROVED** [review]
Indeed, nice to use a builder instead

File: iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/AttachNewLineOnLine.java (hunk 1 above)
Comment (miovd) [reply iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/AttachNewLineOnLine.java:59]: Not sure if this is appropriate indeed! I assumed this javadoc was only useful from a user standpoint (not a developer one) but maybe I am wrong. What do you suggest?

Comment (miovd) [reply iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/AttachNewLineOnLine.java]: I deprecated all constructors except for those used in the builders and just added in the javadoc that they will eventually be package private. Do you agree?

**Pull request merged**